src/Silverback.Integration/Messaging/Broker/Broker.cs

```diff
@@ -46,19 +46,17 @@ public virtual IProducer GetProducer(IEndpoint endpoint)
 
         public virtual IConsumer GetConsumer(IEndpoint endpoint)
         {
-            if (IsConnected)
-                throw new InvalidOperationException("The broker is already connected. Disconnect it to get a new consumer.");
-
-            _logger.LogInformation("Creating new consumer for endpoint {endpointName}.", endpoint.Name);
-
-            var consumer = InstantiateConsumer(endpoint, _consumerBehaviors);
-
-            lock (_consumers)
+            return _consumers.GetOrAdd(endpoint, _ =>
```

Comment:
This leads to the possibility to get an existing consumer in KafkaPartitionsAssignedEvent where we could set the offset(s) by calling Acknowledge(offsets). There might still be an issue with EnableAutoCommit config option, since a commit is done only, when EnableAutoCommit is false ...but default is true

Reply:
I think this will break some existing features (like the possibility to create multiple consumers for the same endpoint, to process messages in parallel).
I have to check that but we may need a different approach to preserve the existing functionalities.